File: go/hamming/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("hamming: sequences have no equal length")
	}

	var hammingDistance int

	for i, ch := range a {
		if byte(ch) != b[i] {
			hammingDistance += 1
		}
	}

	return hammingDistance, nil
}

File: python/hamming/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("sequences have no equal length")

    hamming_distance = 0

    for elem_a, elem_b in zip(strand_a, strand_b):
        if elem_a != elem_b:
            hamming_distance += 1

    return hamming_distance
